runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java

```diff
@@ -180,16 +197,18 @@ private void addWidgets()
 			icon.setSpriteId(SpriteID.WIKI_SELECTED);
 			client.setAllWidgetsAreOpTargetable(true);
 		});
-		icon.setAction(5, "Search"); // Start at option 5 so the target op is ontop
+
+		// set position to 5 if we want Lookup on left-click, or set position to 4 if we want Search
+		final int position = config.leftClickSearch() ? 4 : 5;
+		icon.setAction(position, "Search");
 		icon.setOnOpListener((JavaScriptCallback) ev ->
 		{
-			switch (ev.getOp())
+			if (ev.getOp() == position + 1)
```

Comment:
I don't fully understand why this is `position + 1` but it was `case 6` earlier so i guess it's fine

Reply:
javascriptcallback lives in cs2 land with 1 based indexes, setAction is in engine land with 0 based indexes